src/rbd_fuse/rbd-fuse.cc

```diff
@@ -879,11 +867,36 @@ connect_to_cluster(rados_t *pcluster)
 
 int main(int argc, char *argv[])
 {
+	int r;
 	struct fuse_args args = FUSE_ARGS_INIT(argc, argv);
 
 	if (fuse_opt_parse(&args, &rbd_options, rbdfs_opts, rbdfs_opt_proc) == -1) {
 		exit(1);
 	}
 
+	r = rados_create(&cluster, NULL);
+	if (r < 0) {
+		simple_err("Could not create cluster handle", r);
+		return r;
+	}
+
+	r = rados_conf_parse_env(cluster, NULL);
+	if (r < 0) {
+		simple_err("Error parsing the environment.", r);
+		return -1;
+	}
+
+	r = rados_conf_parse_argv(cluster, argc, (const char **) argv);
```

Comment:
I don't think it is useful as is.

It could be useful, because it would allow to pass common ceph configuration options, like `--id`, ` --cluster`, similarly to what other ceph tools, like `rbd` or `rbd-nbd`, do. The problem here though that earlier `argv` is passed to `fuse_opt_parse`, and I suppose it will fail when it see an unknown (to it) option.

So I see two ways:

1) (simpler) Just don't use `rados_conf_parse_argv` here as it is not useful -- your patch will fix the problem with the config, and this is already an improvement.
2) (more correct in my opinion) I see rados API additionally provides this function:
```
// like above, but return the remainder of argv to contain remaining
// unparsed args.  Must be allocated to at least argc by caller.
// remargv will contain n <= argc pointers to original argv[], the end
// of which may be NULL

extern "C" int rados_conf_parse_argv_remainder(rados_t cluster, int argc,
                                               const char **argv,
                                               const char **remargv)
```
It looks like it is for the cases like this. If we used  `rados_conf_parse_argv_remainder` instead of `rados_conf_parse_argv` and did this before `FUSE_ARGS_INIT`, and then passed `remargv` to `FUSE_ARGS_INIT` instead of `argv` then I suppose it would do the trick.

@rishabh-d-dave  do you want to try (2)? Or you can just provide (1) leaving (2) to be implemented later.

Reply:
I will try (2), since it looks more proper.   :)

And will incorporate the rest of improvements you pointed out.